receiver/mongodbatlasreceiver/logs.go

```diff
@@ -0,0 +1,306 @@
+			}
+		}
+	}
+}
+
+func filterClusters(clusters []mongodbatlas.Cluster, clusterNames []string, include bool) []mongodbatlas.Cluster {
+	// Arrange cluster names for quick reference
+	clusterNameSet := map[string]struct{}{}
+	for _, clusterName := range clusterNames {
+		clusterNameSet[clusterName] = struct{}{}
+	}
+
+	var filtered []mongodbatlas.Cluster
+	for _, cluster := range clusters {
+		if _, ok := clusterNameSet[cluster.Name]; (!ok && !include) || (ok && include) {
+			filtered = append(filtered, cluster)
+		}
+	}
+	return filtered
+}
+
+func (s *logsReceiver) getHostLogs(groupID, hostname, logName string, start, end time.Time) ([]model.LogEntry, error) {
+	// Get gzip bytes buffer from API
+	buf, err := s.client.GetLogs(context.Background(), groupID, hostname, logName, start, end)
+	if err != nil {
+		return nil, err
+	}
+	// Pass this into a gzip reader for decoding
+	reader, err := gzip.NewReader(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	// Logs are in JSON format so create a JSON decoder to process them
+	dec := json.NewDecoder(reader)
+
+	entries := make([]model.LogEntry, 0)
+	for {
+		var entry model.LogEntry
+		err := dec.Decode(&entry)
+		if errors.Is(err, io.EOF) {
+			return entries, nil
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		entries = append(entries, entry)
+	}
+}
+
+func (s *logsReceiver) getHostAuditLogs(groupID, hostname, logName string, start, end time.Time) ([]model.AuditLog, error) {
+	// Get gzip bytes buffer from API
+	buf, err := s.client.GetLogs(context.Background(), groupID, hostname, logName, start, end)
+	if err != nil {
+		return nil, err
+	}
+	reader, err := gzip.NewReader(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	dec := json.NewDecoder(reader)
+
+	entries := make([]model.AuditLog, 0)
+	for {
+		var entry model.AuditLog
+		err := dec.Decode(&entry)
+		if errors.Is(err, io.EOF) {
+			return entries, nil
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		entries = append(entries, entry)
+	}
+}
+
+func (s *logsReceiver) collectLogs(pc ProjectContext, logName, hostname, clusterName string) {
+	logs, err := s.getHostLogs(pc.projectID, hostname, logName, pc.start, pc.end)
+	if err != nil && !errors.Is(err, io.EOF) {
+		s.log.Warn("Failed to retrieve logs from: "+logName, zap.Error(err))
+	}
+
+	for _, log := range logs {
+		plog := mongodbEventToLogData(s.log,
+			log,
+			pc,
+			hostname,
+			clusterName,
+			logName)
+		err := s.consumer.ConsumeLogs(context.Background(), plog)
+		if err != nil {
+			s.log.Error("Failed to consume logs", zap.Error(err))
+		}
+	}
+}
+
+func (s *logsReceiver) collectAuditLogs(pc ProjectContext, logName, hostname, clusterName string) {
+	logs, err := s.getHostAuditLogs(
+		pc.projectID,
+		hostname,
+		logName,
+		pc.start,
+		pc.end)
+
+	if err != nil && !errors.Is(err, io.EOF) {
+		s.log.Warn("Failed to retrieve audit logs: "+logName, zap.Error(err))
+	}
+
+	for _, log := range logs {
+		plog := mongodbAuditEventToLogData(s.log,
```

Comment:
We don't want to create a separate `plog.Logs` for each log entry. They should be grouped by resource. 

I believe at this point in the code all these logs belong to a single resource. So, pass in the whole slice, set up the resource attributes, and then loop through the logs and add them all.